src/gleam/regex.gleam

```diff
@@ -3,8 +3,26 @@
 //// all of the PCRE library is interfaced and some parts of the library go beyond
 //// what PCRE offers. Currently PCRE version 8.40 (release date 2017-01-11) is used.
 
+import gleam/option.{Option}
+
 pub external type Regex
 
+/// The details about a particular match:
+///
+/// - match — the full string of the match.
+/// - index — the index of the match in the original string.
```

Comment:
Is this grapheme index or byte index?

Reply:
Good question, discovered a bug when testing for this.
I assume we want it to be a grapheme index?

I am not sure why Elm thinks some graphemes have a length of 2:
```elm
> Regex.find re "!Gleam"
[{ index = 1, match = "Gleam", number = 1, submatches = [] }]
    : List Regex.Match
> Regex.find re "हGleam"
[{ index = 1, match = "Gleam", number = 1, submatches = [] }]
    : List Regex.Match
> Regex.find re "𐍈Gleam"
[{ index = 2, match = "Gleam", number = 1, submatches = [] }]
    : List Regex.Match
```

Reply:
Elm compiles to JavaScript, and in JS, strings are encoded in UTF-16. I suppose those graphemes are made of two UTF-16 code units.

Reply:
In Elixir it's byte indexed:

```elixir
iex(1)> Regex.scan(~r/Gl\w+/, "!Gleam", return: :index)
[[{1, 5}]]
iex(2)> Regex.scan(~r/Gl\w+/, "हGleam", return: :index)
[[{3, 5}]]
iex(3)> Regex.scan(~r/Gl\w+/, "𐍈Gleam", return: :index)
[[{4, 5}]]
```

Reply:
So I think byte indexed makes the most sense (also from a perfomance point of view)?. And `bit_string.part` etc. can be used for further processing.

Reply:
Bytes makes sense to me

Reply:
Sorry, could you add a note to the docs saying this is a byte index? Possibly rename to `byte_index`

Reply:
Changed the docs that it means byte index.
I did not rename to `byte_index` because in that case I'd feel the need to rename `index` in `FromStringError` as well: https://github.com/gleam-lang/stdlib/blob/2066a256f97583444e16a1e9ddf8fe413834440c/src/gleam/regex.gleam#L25